cumulus/client/consensus/aura/src/collators/basic.rs

```diff
@@ -192,6 +194,20 @@ where
 				Err(e) => reject_with_error!(e),
 			};
 
+			// With async backing this function will be called every relay chain block.
+			//
+			// Most parachains currently run with 12 seconds slots and thus, they would try to
+			// produce multiple blocks per slot which very likely would fail on chain. Thus, we have
+			// this "hack" to only produce on block per slot.
+			//
+			// With https://github.com/paritytech/polkadot-sdk/issues/3168 this implementation will be
+			// obsolete and also the underlying issue will be fixed.
+			if last_processed_slot >= *claim.slot() {
+				continue
+			} else {
+				last_processed_slot = *claim.slot();
```

Comment:
Can we move this assignment at the end of the loops, so that we don't save the slot in case we have errors down the road and let the next block try as well.

Just want to be extra safe and make sure we don't end up in a situation where at first slot pass we don't build the block because runtime doesn't accept it and then at second pass we don't even try anymore.